### ISSUE TITLE
When the API returns a 429, the rejected value is a string

### ISSUE THREAD

**Issue by sdelpuerto-intelygenz** [simplyspoke/node-harvest#109]
When you reach the rate limit, the API returns an object like `{retry_after: 15}`, but with this library the error becomes a string like `Retry after: 15`, which forces me to either use a regular expression to get the seconds to wait, or assume it's always going to be 15. I think neither is optimal, ideally I would like to get the value like so: `const seconds = err.retry_after`.

Is this something you would be willing to change or accept a pull request for?

Thanks in advance.

**Comment by sdelpuerto-intelygenz** [comment]
Closely related to this, I've seen the following code in `client.ts`:

```
preprocess(body, response) {
    // if (headers['retry-after']) {
    //   this.retryAfter(task, headers['retry-after'], done);
    // }

    return { headers: response.headers, data: body };
  }
```

Is there a reason why it's commented? When I was trying to library I saw that promises were simply rejected but not retried and so I implemented my own solution. If you were having trouble with this, I'm more than happy to look into this and try to find a solution.

**Comment by simplyspoke** [comment]
I am happy to add this or work with a provided PR if you have an approach you would like to go with.

If I think back, I believe this was more do to time constraints. It would probably be easy to add in given the right test designed to hit that limit for verification.

**Comment by sdelpuerto-intelygenz** [comment]
Hi, I have a working solution that doesn't reject the promise until after the retry is successful, so that library users can be confident that when a promise is resolved, it means the request was successful regardless of rate limit.

I'm trying to run the tests with a clean repo but I'm getting this error: `Module <rootDir>/node_modules/ts-jest/preprocessor.js in the transform option was not found.`

Indeed that file doesn't exist, so I tried changing the path to `'<rootDir>/node_modules/ts-jest/dist/preprocess.js`, but now I'm getting this error in all test suites: `TypeError: Jest: a transform must export a `process` function.`

I haven't done anything with TypeScript experience and I don't have time to dig into this at work. Could you please look into it?

Should I make a pull request despite the fact that I couldn't create a test suite to hit the rate limit?

**Comment by simplyspoke** [comment]
I am looking at a clean install now. A PR though is a good way to share your progress.

Might not be merge ready, but would allow for some debugging.

**Comment by simplyspoke** [comment]
Looks like one of the updates to ts-jest supports a different config. Updated the required property and it looks to be good now. Unit and integration tests ran without issue.

Looking forward to reviewing the PR. Thanks for all the help.

**Comment by sdelpuerto-intelygenz** [comment]
Thanks for the update. I had to update the tests to reflect my changes, but they work again. I created a PR, not much had to be changed in the end.

Thank **you** for the library :)

**Comment by simplyspoke** [comment]
Your PR works, I can test it locally, but I am having some issues with the pipeline for deployment. Will likely resolve those tomorrow.

**Comment by sdelpuerto-intelygenz** [comment]
Yeah, I noticed. I think it has to do with the environment variables not being set.

**Comment by simplyspoke** [comment]
There we go,

Three parter with an outage, bad data logged in the testing account, and some types causing issues. 

> A new version of the package harvest (2.2.0) was published at 2018-08-09T05:43:43.582Z from
35.224.112.202. The shasum of this package was e04327add8141687c10726cf1a8541e4c78204f8.
